aten/src/ATen/native/quantized/cpu/qlinear_dynamic.cpp

```diff
@@ -204,17 +209,186 @@ class QLinearDynamicInt8 final : public torch::OperatorKernel {
 
     return output;
   }
-#else // USE_FBGEMM
-  at::Tensor operator()(
-      at::Tensor /* input */,
-      at::Tensor /* packed_weight */) {
-    // We make a strong guarantee that models using these operators will have
-    // the same numerics across different machines. Therefore, we do not provide
-    // a fallback path and rather fail loudly if we cannot run FBGEMM.
+#endif // USE_FBGEMM
+#ifdef USE_PYTORCH_QNNPACK
+
+  struct qparams {
+    float scale;
+    std::int32_t zero_point;
+  };
+
+  qparams chooseQuantizationParams(
```

Comment:
This function needs to be identical to the FBGEMM implementation so that numerics are consistent between server and mobile. Is there a reason we cannot reuse the same code?

Reply:
Yes, it is using most of the logic from there. However the code for fbgemm is not available on mobile so we have to repurpose it here for our use case.

Reply:
This is not a aten op, so wondering if it is possible to have a test that checks this implementation for consistency with fbgemm::choose_quantization_params

Reply:
@raghuramank100 Thanks, I'll work on sharing the implementation.